bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairBaseBridgeHandler.java

```diff
@@ -99,355 +143,317 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     }
 
-    /**
-     * Job to send pump query status packages to all Intelliflo Pump things in order to see the status.
-     * Note: From the internet is seems some FW versions of EasyTouch controllers send this automatically and this the
-     * pump status packets can just be snooped, however my controller version does not do this. No harm in sending.
-     *
-     * @author Jeff James
-     *
-     */
-    class PumpStatus implements Runnable {
-        @Override
-        public void run() {
-            List<Thing> things = getThing().getThings();
+    private void internalDisconnect() {
+        if (parserThread != null) {
+            try {
+                parserThread.interrupt();
+                parserThread.join(3000); // wait for thread to complete
+            } catch (InterruptedException e) {
+                // do nothing
+            }
+            parserThread = null;
+        }
 
-            // FF 00 FF A5 00 60 10 07 00 01 1C
-            byte[] packet = { (byte) 0xA5, (byte) 0x00, (byte) 0x00, (byte) id, (byte) 0x07, (byte) 0x00 };
+        reader.ifPresent(close -> {
+            try {
+                close.close();
+            } catch (IOException e) {
+                logger.debug("setInputStream: Exception error while closing: {}", e.getMessage());
+            }
+        });
 
-            PentairPacket p = new PentairPacket(packet);
+        writer.ifPresent(close -> {
+            try {
+                close.close();
+            } catch (IOException e) {
+                logger.debug("setOutputStream: Exception error while closing: {}", e.getMessage());
+            }
+        });
 
-            for (Thing t : things) {
-                if (!t.getThingTypeUID().equals(INTELLIFLO_THING_TYPE)) {
-                    continue;
-                }
+        disconnect();
+
+        connectstate = ConnectState.DISCONNECTED;
+    }
 
-                p.setDest(((PentairIntelliFloHandler) t.getHandler()).id);
-                writePacket(p);
-                try {
-                    Thread.sleep(300); // make sure each pump has time to respond
-                } catch (InterruptedException e) {
-                    break;
+    // Job to pull to try and reconnect upon being disconnected. Note this should only be started on an initial
+    class MonitorIO implements Runnable {
+        @Override
+        public void run() {
+            logger.debug("MonitorIO");
+            // ConnectState.DISCONNECTED implies the connection had at one time been successfully connected and
+            // therefore the
+            // configuration is correct. This state can be reached when an exception happens where the connection has
+            // been
+            // interrupted.
+            if (connectstate == ConnectState.DISCONNECTED || connectstate == ConnectState.CONFIGERROR) {
+                internalConnect();
+            } else if (connectstate == ConnectState.CONNECTED) {
```

Comment:
You should take advantage of using a switch on the `connectstate` enum to simplify this logic.